nixos/lib/test-driver/Machine.pm

```diff
@@ -572,6 +572,12 @@ sub waitForText {
     my ($self, $regexp) = @_;
     $self->nest("waiting for $regexp to appear on the screen", sub {
         retry sub {
+            my ($retries_remaining) = @_;
+            if ($retries_remaining == 0) {
```

Comment:
Just a note: Instead of unpacking the array, you could simply use `$_[0]` here as well.